.travis.yml

```diff
@@ -1,14 +1,28 @@
 language: php
 
-php:
-  - 5.3
-  - 5.4
-  - 5.5
-  - 5.6
-  - hhvm
+matrix:
+    include:
+        - php: 5.3
+        - php: 5.4
+        - php: 5.5
+        - php: 5.6
+        - php: 5.3
+          env: deps=low
+        - php: 5.6
+          env: deps=high
+        - php: hhvm
 
-before_script: 
-    - composer install --dev --prefer-source
+env:
+    global:
+        - deps=no
+
+before_install:
+    - composer self-update
+
+install:
+    - if [ "$deps" = "no" ]; then composer --prefer-source install; fi;
+    - if [ "$deps" = "low" ]; then composer --prefer-source --prefer-lowest --prefer-stable update; fi;
+    - if [ "$deps" = "high" ]; then composer --prefer-source update; fi;
```

Comment:
deps=no or deps=high is doing exactly the same here, given there is no lock file. There is no need to duplicate things IMO